src/data/finsupp.lean

```diff
@@ -463,6 +463,22 @@ lemma prod_ite_eq' [has_zero β] [comm_monoid γ] (f : α →₀ β) (a : α) (b
   f.prod (λ x v, ite (x = a) (b x v) 1) = ite (a ∈ f.support) (b a (f a)) 1 :=
 by { dsimp [finsupp.prod], rw f.support.prod_ite_eq', }
 
+@[simp] lemma prod_pow [fintype α] [comm_monoid γ] (f : α →₀ ℕ) (g : α → γ) :
+  f.prod (λ a b, g a ^ b) = ∏ a, g a ^ (f a) :=
+begin
+  classical,
+  calc f.prod (λ a b, g a ^ b) = ∏ a, (if a ∈ f.support then g a ^ f a else 1) : _
+    ... = _ : _,
+  { rw [finsupp.prod, ← finset.prod_filter, finset.prod_congr],
```

Comment:
Isn't this `(prod_extend_by_one f.support _).symm`?

Reply:
And `prod_pow` looks like `prod_subset`.

Reply:
It is, but I wasn't sure if we wanted to import `data.fintype.card` into this file. So I reproved it.

Reply:
Thanks, `prod_subset` led to serious golfing :wink: